lib/mbedtls/sha256.c

```diff
@@ -60,3 +61,40 @@ void sha256_csum_wd(const unsigned char *input, unsigned int ilen,
 
 	sha256_finish(&ctx, output);
 }
+
+void sha256_hmac(const unsigned char *key, int keylen,
```

Comment:
This function seems to be generic and should be located in lib instead of lib/mbedtls